server/src/test/java/org/elasticsearch/action/admin/indices/settings/put/UpdateSettingsRequestTests.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.indices.settings.put;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.settings.Settings.Builder;
+import org.elasticsearch.common.util.CollectionUtils;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.test.AbstractStreamableTestCase;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.function.Predicate;
+
+import static org.elasticsearch.test.XContentTestUtils.insertRandomFields;
+import static org.hamcrest.Matchers.equalTo;
+
+public class UpdateSettingsRequestTests extends AbstractStreamableTestCase<UpdateSettingsRequest> {
+
+    @Override
+    protected UpdateSettingsRequest mutateInstance(UpdateSettingsRequest request) {
+        if (randomBoolean()) {
+            return new UpdateSettingsRequest(mutateSettings(request.settings()), request.indices());
+        }
+        return new UpdateSettingsRequest(request.settings(), mutateIndices(request.indices()));
+    }
+
+    @Override
+    protected UpdateSettingsRequest createTestInstance() {
+        return randomBoolean()
+                ? new UpdateSettingsRequest(randomSettings(0, 2))
+                : new UpdateSettingsRequest(randomSettings(0, 2), randomIndicesNames(0, 2));
+    }
+
+    @Override
+    protected UpdateSettingsRequest createBlankInstance() {
+        return new UpdateSettingsRequest();
+    }
+
+    @Override
+    protected void assertEqualInstances(UpdateSettingsRequest expectedInstance, UpdateSettingsRequest newInstance) {
+        newInstance.indices(expectedInstance.indices());
+        super.assertEqualInstances(expectedInstance, newInstance);
+    }
+
+    public void testXContent() throws IOException {
```

Comment:
do you think that it would be possible to have two different test classes, one for the xcontent bits, that doesn't rely on equals/hashcode , and the other one like this on that we have that tests serialization and ordinary equals/hashcode ? Let me know if I am missing something.